src/components/People/PeoplePage.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useContext } from 'react';
+import { Link } from 'react-router-dom';
+import cn from 'classnames';
+import { Loader } from '../Loader';
+import { PeopleContext } from '../../context/PeopleContext';
+import {
+  selectPerson, DispatchFunction,
+} from '../../context/PeopleActions';
+import { Person } from '../../types';
+import { ErrorType } from '../../types/ErrorType';
+
+export const PeoplePage: React.FC = () => {
+  const { dispatch } = useContext(PeopleContext);
+  const { state } = useContext(PeopleContext);
+  const {
+    isLoading, people, selectedPerson, error,
+  } = state;
+
+  const createParentLink = (
+    parentName: string | null,
+    peopleArray: Person[],
+    localDispatch: DispatchFunction,
+  ) => {
+    const parent = peopleArray.find(p => p.name === parentName);
+    const parentSlug = parent ? parent.slug : null;
+    const className = parent?.sex === 'f' ? 'has-text-danger' : '';
+
+    return parentSlug ? (
+      <Link
+        to={`/people/${parentSlug}`}
+        onClick={() => localDispatch(
+          { type: 'SELECT_PERSON', payload: parentSlug },
+        )}
+        className={className}
+      >
+        {parentName}
+      </Link>
+    ) : parentName;
+  };
```

Comment:
This function looks like a separate component. So you can formally move this code to the separate component (e.g. `ParentLink`). It will look better.